### PR TITLE
chore/ unit-test-registration

### PR DESCRIPTION
**Pull request by mikco-hipe**
### OVERVIEW

Added unit tests for Registration page

### SNIPPETS

image

### CODE REVIEW

**Review by bpocrafael — APPROVED** [review]

File: laravel-app/tests/Unit/RegisterControllerTest.php [hunk 1]
```diff
@@ -7,15 +7,58 @@
...
+            'email_verification_code' => Str::random(40),
+        ]);
+
+        $user->information()->create([
+            'first_name' => $userData['first_name'],
+            'last_name' => $userData['last_name'],
+        ]);
+
+        Mail::to($userData['email'])->send(new EmailVerificationMail($user));
+    }
```
Comment (hipe01mark) [inline laravel-app/tests/Unit/RegisterControllerTest.php:51]: Why is your test not testing your service methods? non-sense right?? you are just testing the default creation of laravel.

File: laravel-app/tests/Unit/RegisterControllerTest.php [hunk 2]
```diff
@@ -49,4 +96,155 @@ public function test_invalid_register_controller_test()
...
+
+        $request = new CreateUserRequest();
+        $validator = validator($data, $request->rules(), $request->messages());
+
+        $this->assertFalse($validator->fails());
+
+        $this->registerUser($data);
+
+        $this->assertEmailSent($data['email']);
+    }
```
Comment (hipe01mark) [inline laravel-app/tests/Unit/RegisterControllerTest.php:119]: Why is your test not testing your service methods? non-sense right?? you are just testing the default way of creation data in laravel.

File: laravel-app/tests/Unit/RegisterControllerTest.php [hunk 3]
```diff
@@ -49,4 +96,155 @@ public function test_invalid_register_controller_test()
...
+
+        $responseExistingEmail = $this->post('/register', [
+            'name' => 'Another User',
+            'email' => 'existing@example.com',
+            'password' => 'password',
+            'password_confirmation' => 'password',
+        ]);
+
+        $responseExistingEmail->assertSessionMissing('status');
+    }
```
Comment (hipe01mark) [inline laravel-app/tests/Unit/RegisterControllerTest.php:141]: Why is your test not testing your service methods? non-sense right?? you are just testing the default way of creation data in laravel.

File: laravel-app/tests/Unit/RegisterControllerTest.php [hunk 4]
```diff
@@ -49,4 +96,155 @@ public function test_invalid_register_controller_test()
...
+            'password' => bcrypt('password'),
+            'email_verification_code' => 'sample_verification_code',
+        ]);
+
+        $verificationUrl = route('verify-email', ['verification_code' => $user->email_verification_code]);
+
+        $response = $this->get($verificationUrl);
+
+        $response->assertRedirect('/login');
+    }
```
Comment (hipe01mark) [inline laravel-app/tests/Unit/RegisterControllerTest.php:158]: Why is your test not testing your service methods? non-sense right?? you are just testing the default way of creation data in laravel.

File: laravel-app/tests/Unit/RegisterControllerTest.php [hunk 5]
```diff
@@ -49,4 +96,155 @@ public function test_invalid_register_controller_test()
...
+            'email_verified_at' => null,
+        ]);
+
+        $response = $this->post('/login', [
+            'email' => 'unverified@example.com',
+            'password' => 'password',
+        ]);
+
+        $response->assertRedirect('register/resend');
+    }
```
Comment (hipe01mark) [inline laravel-app/tests/Unit/RegisterControllerTest.php:176]: Why is your test not testing your service methods? non-sense right?? you are just testing the default way of creation data in laravel.

File: laravel-app/tests/Unit/RegisterControllerTest.php [hunk 6]
```diff
@@ -49,4 +96,155 @@ public function test_invalid_register_controller_test()
...
+        $response = $this->post('/register/resend', [
+            'email-resend' => 'unverified@example.com',
+        ]);
+
+        Mail::assertSent(EmailVerificationMail::class, function ($mail) use ($user) {
+            return $mail->hasTo($user->email);
+        });
+
+        $response->assertRedirect('/login');
+    }
```
Comment (hipe01mark) [inline laravel-app/tests/Unit/RegisterControllerTest.php:209]: Why is your test not testing your service methods? non-sense right?? you are just testing the default way of creation data in laravel.

File: laravel-app/tests/Unit/RegisterControllerTest.php [hunk 7]
```diff
@@ -49,4 +96,155 @@ public function test_invalid_register_controller_test()
...
+
+        $responseVerified->assertSessionHas('error', 'Email doesn\'t exist in the database or is already verified');
+
+        $responseNonExistent = $this->post('/register/resend', [
+            'email-resend' => 'nonexistent@example.com',
+        ]);
+
+        $responseNonExistent->assertSessionHas('error', 'Email doesn\'t exist in the database or is already verified');
+    }
+
```
Comment (hipe01mark) [inline laravel-app/tests/Unit/RegisterControllerTest.php:233]: Why is your test not testing your service methods? non-sense right?? you are just testing the default way of creation data in laravel.

File: laravel-app/tests/Unit/RegisterControllerTest.php [hunk 8]
```diff
@@ -49,4 +96,155 @@ public function test_invalid_register_controller_test()
...
+            'password' => bcrypt('password'),
+            'email_verification_code' => 'verification_code',
+            'email_verified_at' => null,
+        ]);
+
+        $response = $this->get(route('verify-email', ['verification_code' => 'sample_outdated_verification_code']));
+
+        $response->assertRedirect(route('resend'));
+        $response->assertSessionHas('error', 'Verification link expired or email already verified');
+    }
```
Comment (hipe01mark) [inline laravel-app/tests/Unit/RegisterControllerTest.php:249]: Why is your test not testing your service methods? non-sense right?? you are just testing the default way of creation data in laravel.

File: laravel-app/tests/Unit/RegisterControllerTest.php (hunk 1 above)
Comment (mikco-hipe) [reply laravel-app/tests/Unit/RegisterControllerTest.php:51]: Apologies sir 🙇 will revise it now 🙇

File: laravel-app/tests/Unit/RegisterControllerTest.php (hunk 2 above)
Comment (mikco-hipe) [reply laravel-app/tests/Unit/RegisterControllerTest.php:119]: Apologies sir 🙇 will revise it now 🙇

File: laravel-app/tests/Unit/RegisterControllerTest.php (hunk 3 above)
Comment (mikco-hipe) [reply laravel-app/tests/Unit/RegisterControllerTest.php:141]: Apologies sir 🙇 will revise it now 🙇

File: laravel-app/tests/Unit/RegisterControllerTest.php (hunk 4 above)
Comment (mikco-hipe) [reply laravel-app/tests/Unit/RegisterControllerTest.php:158]: Apologies sir 🙇 will revise it now 🙇

File: laravel-app/tests/Unit/RegisterControllerTest.php (hunk 5 above)
Comment (mikco-hipe) [reply laravel-app/tests/Unit/RegisterControllerTest.php:176]: Apologies sir 🙇 will revise it now 🙇

File: laravel-app/tests/Unit/RegisterControllerTest.php (hunk 6 above)
Comment (mikco-hipe) [reply laravel-app/tests/Unit/RegisterControllerTest.php:209]: Apologies sir 🙇 will revise it now 🙇

File: laravel-app/tests/Unit/RegisterControllerTest.php (hunk 7 above)
Comment (mikco-hipe) [reply laravel-app/tests/Unit/RegisterControllerTest.php:233]: Apologies sir 🙇 will revise it now 🙇

File: laravel-app/tests/Unit/RegisterControllerTest.php (hunk 8 above)
Comment (mikco-hipe) [reply laravel-app/tests/Unit/RegisterControllerTest.php:249]: Apologies sir 🙇 will revise it now 🙇

**Review by bpocrafael — APPROVED** [review]

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request merged**